### ISSUE TITLE
change language err

### ISSUE THREAD

**Issue by moddc** [typemill/typemill#248]
you cannot change the language pack. Preference is saved, but the language is not changed.
Tested on two different hosts.
chmod write issued

**Comment by trendschau** [comment]
thank you for reporting the error, will try to fix with next release.

**Comment by hide-me** [comment]
Hmmm... For several days I have been working on improving my translation and adding new strings to language files. But I have not seen problems with switching languages.

Of course, this does not mean that there is no bug. However, I think we need more detailed information about both hosts and TypeMill version.

**Comment by trendschau** [comment]
it seems that I have messed up the translation with the cleanup in version 1.4.4 or 1.4.3, I get the same error with a fresh live installation under a subdomain like mysub.main.com.

I think this change in system.php line 271 will fix it:

    $pieces = explode('/',trim($uri->getPath(),'/'));

Sorry for that. The cleanup was a bit more urgent and complex (proxy detection) so I did not had the chance to test translations deeply.

will try to publish it as a hotfix next weekend.
